New PR - Audit & fix Information.md

Yoda-Canada opened the pull request:
I closed the PR #81 and open the new PR for Audit & fix Information.md page #44

Steps done to modify and check Information.md:

1. Fixed images work in dark modes. The memory model image's downward arrows and the letter in the bottom right corner are not clear in dark modes. I wrapping them in a <span> tag and adding class "mdImg" to it. 
2. The images can't display in all models. Change all the image path from "/img/.jpg" to "../../static/img/.jpg". The issue is fixed.
3. Fix Frontmatter for the page to include id, title, slug, and description.
4. Replace blockquote with Admonitions for clarity.
5. Added alt text for diagram images.
6. Some of " * "will get converted into _ after being formatted with Prettier.  I use \* replace them.

hlavu commented on docs/A-Introduction/information.md:
```diff
@@ -78,33 +87,44 @@ To learn how to convert between hexadecimal and binary refer to the chapter enti
 
-* consists of 32 (= 4 * 8) Gigabits
+* consists of 32 (= 4 \* 8) Gigabits
 * starts at a low address of 0x00000000
 * ends at a high address of 0xFFFFFFFF
```
I think address values in these lines should be wrapped in backticks.

Yoda-Canada replied:
I have modified it, thx.

humphd replied:
You haven't.  Please make the correction.

Yoda-Canada replied:
I have modified, thx

lmpham1 commented on docs/A-Introduction/information.md:
```diff
@@ -78,33 +87,44 @@ To learn how to convert between hexadecimal and binary refer to the chapter enti
+* Peta or P (=1024T): 1 Petabyte = 1024 \* 1024 \* 1024 \* 1024 \* 1024 bytes ~ 10<sup>15</sup> bytes
+* Exa or E (=1024P): 1 Exabyte = 1024 \* 1024 \* 1024 \* 1024 \* 1024 \* 1024 bytes ~ 10<sup>18</sup> bytes
 
 Note that the multiplying factor is 1024, not 1000.  1024 bytes is 2<sup>10</sup> bytes, which is approximately 10<sup>3</sup> bytes.
```
I think this line can be wrapped in admonition as well

humphd replied:
This needs to get done still.

humphd commented on docs/A-Introduction/information.md:
```diff
@@ -78,33 +87,44 @@ To learn how to convert between hexadecimal and binary refer to the chapter enti
 
+<span className="mdImg" align="center" alt="A memory model image">
+
 ![RAM memory model: linearly mapped \(continuous sequence\)](https://ict.senecacollege.ca//~ipc144/pages/images/ram.png)
```
This image needs to be brought into the repo

Yoda-Canada replied:
I have modified it, thx.

humphd replied:
I don't understand, it's still pointing to https://ict.senecacollege.ca//~ipc144/pages/images/ram.png.  All images need to get imported into this repo and added to git.

humphd reviewed (changes requested):
Please don't say you have done things if they are not done.  Let me know when these changes have been pushed.

Yoda-Canada commented:
@humphd I see. Next time, I will upload all of them successfully and then reply.  I still confuse the rebase command, and still can't merge all commits.
1. Yesterday, I create a new branch named issue-44 and modified all contents in this branch. I rebase all commits and uploaded with one commit successfully.
2. Today, I find there are new reviews, then I modified them and rebase all commits to one successfully.
3. I use the command "git push origin issue-44". The upload failed because lots of new commits ahead of this commit.
4. I use " git checkout main" -> "git pull upstream main" -> "git merge issue-44" -> "git push origin main". It uploaded successfully. The PR #101 can't Synchronization Update because it PR from issue-44 branch.
5. Then I "git pull origin issue-44". There are a lot of new commits from upstream and they mixed with my commit. I searched some methods from the website, but it still can't be one. 
Now all reviews are modified and updated, but 6 commits can't be one. I have no idea how to deal with it. Maybe it would be better to create a new one.

TueNguyen2911 commented:
@Yoda-Canada. 
4. You should have merged the `main` branch into `issue-44` and resolved any conflicts and then pushed it to automatically update your PR
5. 
- To fix this, I think you can try is going to your `issue-14`, use `git reflog`, check [here](https://medium.com/@shreyawhiz/how-to-undo-a-mistaken-git-rebase-life-saver-2977ff0a0602) for more information. Revert your squashed commit by using `git reset — hard <commit sha hash>`, make changes, `rebase` and push it to issue-44 again. Before doing this, you can checkout a new branch to be extra safe
- If it doesn't fix, you can try cloning your project again and repeat the instruction above

Yoda-Canada commented:
> @Yoda-Canada. 4. You should have merged the `main` branch into `issue-44` and resolved any conflicts and then pushed it to automatically update your PR 5.
> 
> * To fix this, I think you can try is going to your `issue-14`, use `git reflog`, check [here](https://medium.com/@shreyawhiz/how-to-undo-a-mistaken-git-rebase-life-saver-2977ff0a0602) for more information. Revert your squashed commit by using `git reset — hard <commit sha hash>`, make changes, `rebase` and push it to issue-44 again. Before doing this, you can checkout a new branch to be extra safe
> * If it doesn't fix, you can try cloning your project again and repeat the instruction above

@TueNguyen2911 Thank you so much, but it can't resolve the issue.

TueNguyen2911 commented:
@Yoda-Canada, I'll add yours as `remote` to my local and try to resolve this issue, I'll get back to you if I succeed

Yoda-Canada commented:
> 

@TueNguyen2911 oh, I have already delete the rpo

humphd commented:
@Yoda-Canada have you opened a third PR for this?  Please take some time to figure out what you're doing wrong with git so this isn't necessary.  You should never need to delete a repo.

humphd commented:
Closing in favour of #105.  I think all the comments here are addressed there.
